enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+- Have a generic component to consume placements and generate placement decisions;
+- Limit the access to managed clusters by leveraging ManagedClusterSets;
+- Replace and deprecate the current placementrule APIs in https://github.com/open-cluster-management/multicloud-operators-placementrule
+
+### Non-Goals
+
+- Support tolerations/taints (user story 8). ManagedCluster has no taint defined in spec yet;
+- Create metrics data when creating/updating placement decisions (user story 9). It needs further investigation and should be covered in a separate enhancement;
+- Support time window (user story 10). It should be covered in a separate enhancement;
+
+## Proposal
+
+### API Spec
+
+#### Placement API
+`Placement` is a namespace scoped API and represents a selection of clusters in one or multiple ManagedClusterSets. When a placement is created, a controller consuming the placement should generate one or multiple placement decisions based on the spec defined in placement.
+```go
+// Placement defines a rule to select a set of ManagedClusters from the ManagedClusterSets bound
+// to the placement namespace.
+//
+// Here is how the placement policy combines with other selection methods to determine a matching
+// list of clusters:
+// 1) Clusters are registered with hub as cluster-scoped ManagedClusters;
+// 2) ManagedClusters are organized into cluster-scoped ManagedClusterSets;
+// 3) ManagedClusterSets are bound to workload namespaces;
+// 4) Namespace-scoped Placements specify a slice of ManagedClusterSets which select a working set
+//    of potential ManagedClusters;
+// 5) Then Placements subselect from that working set using label/claim selection.
+//
+// No ManagedClusters will be selected if there is no ManagedClusterSet bound to the placement
+// namespace. User is able to bind a ManagedClusterSet to a namespace by creating a 
+// ManagedClusterSetBinding in the namespace if they have an RBAC rule to CREATE on the virtual
+// subresource of `managedclustersets/bind`.
+//
+// A slice of PlacementDecisions with label placement.open-cluster-management.io={placement name}
+// will be created to represent the cluster selected by this placement.
+// 
+// If a ManagedCluster is selected and added into the PlacementDecisions, other components may apply
+// workload on it; once it is removed from the PlacementDecisions, the workload applied on this
+// ManagedCluster should be evicted accordingly.
+type Placement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of Placement.
+	Spec PlacementSpec `json:"spec"`
+
+	// Status represents the current status of the Placement
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+// PlacementSpec defines the attributes of Placement.
+// An empty PlacementSpec selects all clusters from the bound ManagedClusterSets to placement namespace.
+type PlacementSpec struct {
```

Comment:
If I want to have my Placement select only "healthy" or "reporting" managedclusters, is there a way to do that?

Reply:
> If I want to have my Placement select only "healthy" or "reporting" managedclusters, is there a way to do that?

If we don't know how to do this now, making it a requirement to consider for beta is fine.

Reply:
We plan to support selecting clusters with toleration/taints when api version moves to beta.  Unhealthy clusters will be marked with some taints which prevent the clusters from being selected by a placement without any toleration specified.